pkg/policy/repository.go

```diff
@@ -137,25 +138,26 @@ type Repository struct {
 	// PolicyCache tracks the selector policies created from this repo
 	policyCache *PolicyCache
 
-	certManager CertificateManager
+	certManager   certificatemanager.CertificateManager
+	secretManager certificatemanager.SecretManager
```

Comment:
FWIW, this struct field is not set in `NewPolicyRepository` below, which leads to TLS policies being broken. The respective test is currently quarantined, thus the issue was not caught on the PR. I've opened https://github.com/cilium/cilium/pull/23895 with a fix.